src/thermo/ThermoFactory.cpp

```diff
@@ -561,7 +561,16 @@ void setupPhase(ThermoPhase& thermo, AnyMap& phaseNode, const AnyMap& rootNode)
     thermo.initThermo();
 
     if (phaseNode.hasKey("state")) {
-        thermo.setState(phaseNode["state"].as<AnyMap>());
+        auto node = phaseNode["state"].as<AnyMap>();
+        if (thermo.isPure() &&
+            (node.hasKey("X") || node.hasKey("mole-fractions") ||
+             node.hasKey("Y") || node.hasKey("mass-fractions")) ) {
+            warn_user("setupPhase",
+                "Invalid specification for stoichiometric phase '{}' ('{}'). "
+                "State specifies concentrations, which for a single species "
+                "phase is unity by definition.", thermo.name(), thermo.type());
+        }
+        thermo.setState(node);
```

Comment:
I think this falls under the previous arguments regarding single-species phases, and that therefore this check should be removed.

Reply:
I agree, but would also say that explicitly specifying the composition of a single species is redundant. I'm suggesting to change the wording instead and leave the warning.

Reply:
@ischoegl But why do users need to be warned at all? Redundant is fine since it doesn't actually cause any problems, and allowing this (without warnings) means that more code can be in common for more phase types.

Reply:
It can be set with 'X' being 0.5, which would be normalized to 1. I am suggesting to leave this in to avoid clutter (or misleading information) in user created input.

Reply:
A user could set `X` to 0.5, which would be misleading (apart from being clutter). Hence my suggestion to leave this in to inform users when they create custom input.

Reply:
Sure, and I think the example @speth gave earlier was that a user could set `X` for an ideal gas phase with one species to 0.5, but that case won't raise a warning even though the result is the same as here. I really don't see the case for warning users about this. It seems like noise, and at some point, users have to be trusted to know what they're doing.

Reply:
Ok. Will take it out.

Reply:
ok.